src/transformers/tokenization_utils_base.py

```diff
@@ -1270,8 +1270,11 @@ def save_pretrained(self, save_directory) -> Tuple[str]:
             This method make sure the full tokenizer can then be re-loaded using the
             :func:`~transformers.PreTrainedTokenizer.from_pretrained` class method.
         """
-        if not os.path.isdir(save_directory):
-            logger.error("Saving directory ({}) should be a directory".format(save_directory))
+
+        if not os.path.exists(save_directory):
+            os.path.makedirs(save_directory)
+        elif os.path.isfile(save_directory):
```

Comment:
curious why those if clauses are reversed vs. the other ones

Reply:
programmer error

Reply:
same here I would just go with the `exists_ok=True` option.